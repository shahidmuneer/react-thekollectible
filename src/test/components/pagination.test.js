import { render } from '../utils'
import Pagination from '../../components/pagination'
import React from 'react'

/**
 *
 * @case
 * - initial list number page [«,‹,1,2,3,4,5,›,»]
 */

describe('<Pagination /> component', () => {
  it('renders null if the props of params[pageNumber] and or totalCount is undefined', () => {
    const wrapper = render(<Pagination />)
    expect(wrapper.container.firstChild).toBeNull()
  })
  it('renders 3 buttons page number', () => {
    const {container, asFragment} = render(<Pagination params={{pageNumber: 1, limit: 5}} totalCount={15} />)
    expect(container.querySelectorAll('#page-number').length).toEqual(3)
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders 5 buttons page number & active class on page 1', () => {
    const pageNumber = 3
    const {container, asFragment} = render(<Pagination params={{pageNumber, limit: 5}} totalCount={50} />)
    expect(container.querySelectorAll('#page-number').length).toEqual(5)
    expect(container.querySelectorAll('#page-number')[pageNumber - 1].classList.contains('active')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})