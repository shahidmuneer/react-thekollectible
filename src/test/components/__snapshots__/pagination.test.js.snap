// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> component renders 3 buttons page number 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="pagination"
    >
      <li
        class="page-item active"
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          1
        </span>
      </li>
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          2
        </span>
      </li>
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          3
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`<Pagination /> component renders 5 buttons page number & active class on page 1 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="pagination"
    >
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          1
        </span>
      </li>
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          2
        </span>
      </li>
      <li
        class="page-item active"
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          3
        </span>
      </li>
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          4
        </span>
      </li>
      <li
        class="page-item "
        id="page-number"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          5
        </span>
      </li>
      <li
        class="page-item "
        id="button-helper"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          ›
        </span>
      </li>
      <li
        class="page-item "
        id="button-helper"
      >
        <span
          class="page-link"
          style="cursor: pointer;"
        >
          »
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
